### PR TITLE
feat: add folder exclusions

### PR DESCRIPTION
**Pull request by mikelaramie**
***Issue***: https://lacework.atlassian.net/browse/ALLY-953

**User Story**
As a customer, I would like to exclude certain folders from my Lacework integration

***Description:***
This is a first pass at adding folder-level exclusions into our terraform logic.  This will address customer requests to exclude things like the `system-gsuite` folders that can grow into a large amount of folders with no cloud resources, leading to a suboptimal UI experience within Lacework.

As an initial release, this code only handles exclusions of root-level folders; it will not consider subfolders below the root of the GCP organization.  It also introduces a flag `include_root_projects` that defaults to `true` - this can be set to `false` if the customer does not want to include the projects that may be located in the root of the organization.

***Additional Info:***
Changes that may impact other parts of the code:
- `local.logging_sink_writer_identity` is now a set of strings (was a string)
- Any logic that checks for `var.org_integration` now checks for the status of `var.exclude_folders` as well

Testing has been performed against:
- Org-level integration with no exclusions
- Org-level integration with multiple exclusions
- Project-level integration

### CODE REVIEW

File: examples/organization-level-audit-log-exclude-folders/README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+provider "lacework" {}
+
+module "gcp_organization_level_audit_log" {
+  source               = "lacework/audit-log/gcp"
+  version              = "~> 1.0"
+  bucket_force_destroy = true
+  org_integration      = true
+  organization_id      = "my-organization-id"
+}
```
Comment (rmoles) [inline examples/organization-level-audit-log-exclude-folders/README.md:23]: It would be good to update this README with an example of the exclude

Comment (mikelaramie) [reply examples/organization-level-audit-log-exclude-folders/README.md]: Helps when you hit save before commiting.  Should be in there now, thanks for catching that!

**Comment by afiune** [comment]
Make it so

**Comment by afiune** [comment]
Make it so! 💯

File: examples/organization-level-audit-log/main.tf [hunk 2]
```diff
@@ -2,6 +2,14 @@ provider "google" {}
 
 provider "lacework" {}
 
+variable "organization_id" {
+  default = "my-organization-id"
+}
+
+variable "project_id" {
```
Comment (afiune) [inline examples/organization-level-audit-log/main.tf:9]: where do we use this variable?

**Comment by afiune** [comment]
make it so

File: examples/organization-level-audit-log/main.tf (hunk 2 above)
Comment (mikelaramie) [reply examples/organization-level-audit-log/main.tf:9]: I need to pass this to the module, good catch

**Review by afiune — CHANGES_REQUESTED** [review]
This change will need a major version bump based of this feedback https://github.com/lacework/terraform-gcp-config/issues/51

**Comment by mikelaramie** [comment]
No pressure to look at this yet - just testing the updates in codefresh.

**Comment by rmoles** [comment]
could we get a secondary ticket created to incorporate this functionality into the new CLI TF generation?

**Comment by afiune** [comment]
make it so

**Comment by afiune** [comment]
@rmoles https://lacework.atlassian.net/browse/ALLY-923

File: versions.tf [hunk 3]
```diff
@@ -2,7 +2,7 @@ terraform {
   required_version = ">= 0.12.31"
```
Comment (afiune) [inline versions.tf:2]:
```suggestion
  required_version = ">= 0.14.0"
```

**Comment by afiune** [comment]
Make it so

File: examples/organization-level-audit-log-exclude-folders/README.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
+# Integrate GCP Organization with Lacework
+The following provides an example of integrating a Google Cloud Organization with Lacework for Cloud Audit Log analysis, excluding 2 folders from the root integration.
```
Comment (afiune) [inline examples/organization-level-audit-log-exclude-folders/README.md:2]:
```suggestion
The following provides an example of integrating a Google Cloud Organization with Lacework for Cloud
Audit Log analysis, excluding 2 folders from the root integration.
```

File: examples/organization-level-audit-log-exclude-folders/README.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  lifecycle_rule_age   = 7
+  exclude_folders      = true
+  folders_to_exclude   = [
+    "folders/578370918314", 
+    "folders/1099205162015",
+  ] 
+}
+```
+
+For detailed information on integrating Lacework with Google Cloud see [GCP Compliance and Audit Trail Integration - Terraform From Any Supported Host](https://support.lacework.com/hc/en-us/articles/360057065094-GCP-Compliance-and-Audit-Trail-Integration-Terraform-From-Any-Supported-Host)
```
Comment (afiune) [inline examples/organization-level-audit-log-exclude-folders/README.md:33]:
```suggestion
For detailed information on integrating Lacework with Google Cloud see [GCP Compliance and Audit Trail Integration - Terraform From Any Supported Host](https://docs.lacework.com/gcp-compliance-and-audit-log-integration-terraform-from-any-supported-host)
```

**Comment by afiune** [comment]
MAKE IT SO!

**Review by afiune — APPROVED** [review]
tenor-127579209

**Pull request merged**